File: gpt-3.5/template_context_v2/wait_clause.c
#include "acc_testsuite.h"
#ifndef T1
// T1: wait clause, V: 2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    #pragma acc data copy(err)
    {
        int x = 0;

        #pragma acc parallel loop
        for (int i = 0; i < VECTOR_LENGTH; ++i){
            x = x + i;
        }

        #pragma acc parallel loop wait
        for (int i = 0; i < VECTOR_LENGTH; ++i){
            err += i;
        }
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Phind-CodeLlama-34B-v2/stage2/wait_clause.f90
#ifndef T1
!T1:wait clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, DIMENSION(LOOPCOUNT) :: a, b, c

        !$acc data copy(a(1:LOOPCOUNT), b(1:LOOPCOUNT), c(1:LOOPCOUNT))
          !$acc parallel loop async(1)
          DO i = 1, LOOPCOUNT
            a(i) = 3.14
          END DO

          !$acc parallel loop async(2)
          DO i = 1, LOOPCOUNT
            b(i) = 42.0
          END DO

          !$acc parallel loop async(3)
          DO i = 1, LOOPCOUNT
            c(i) = a(i) * b(i)
          END DO

          !$acc wait(1, 2, 3)
        !$acc end data

        DO i = 1, LOOPCOUNT
          IF (ABS(c(i) - 128.616) .gt. PRECISION) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM